desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -639,10 +628,17 @@ void DownloadFromDCWidget::enableBluetoothMode(int state)
 {
 	ui.chooseBluetoothDevice->setEnabled(state == Qt::Checked);
 
-	if (state == Qt::Checked)
-		selectRemoteBluetoothDevice();
-	else
-		ui.device->setCurrentIndex(-1);
+	/*	This is convoluted enough to warrant explanation:
+ 		1. If we have a Bluetooth device, but no Bluetooth address then scan.
+		2. If we have a Bluetooth device and address, skip scan.
+		3. If it's not Bluetooth, clear the device address unless it's a Mac.
+		Mac USB addresses tend to be persistent and are better saved. */
+	if (state == Qt::Checked) {
+		if (!isBluetoothAddress(ui.device->currentText()))
+			selectRemoteBluetoothDevice();
+	} else
+		if (isBluetoothAddress(ui.device->currentText()) || QSysInfo::kernelType() != "darwin")
```

Comment:
I did some testing on linux with this, and this is now removing the remembered filesystem path for Garmin devices - so I think it will be better if we remove the ` || QSysInfo::kernelType() != "darwin"` - the fact that we are loading the device from a list of the detected devices with `ui.device->setCurrentIndex(deviceIndex(qPrefDiveComputer::device##num()))` will make sure that we are not setting a non-existent device.

Reply:
Thanks for testing this - removed Mac specific behavior and any non-Bluetooth mount points will be preserved.